txcache/selection.go

```diff
@@ -91,14 +91,18 @@ func selectTransactionsFromBunches(
 
 		shouldSkipTransaction := detectSkippableTransaction(virtualSession, item)
 		if !shouldSkipTransaction {
-			accumulatedGas += gasLimit
 			selectedTransaction := item.selectCurrentTransaction()
-			selectedTransactions = append(selectedTransactions, selectedTransaction)
```

Comment:
For reference, I'll post the current implementation here (Andromeda):

```
shouldSkipTransaction := detectSkippableTransaction(sessionWrapper, item)
if !shouldSkipTransaction {
	accumulatedGas += gasLimit
	selectedTransaction := item.selectCurrentTransaction()
	selectedTransactions = append(selectedTransactions, selectedTransaction)
	sessionWrapper.accumulateConsumedBalance(selectedTransaction)
}
```

https://github.com/multiversx/mx-chain-storage-go/blob/main/txcache/selection.go#L74


`detectSkippableTransaction` should detect an insufficient balance for fees.

`accumulateConsumedBalance` does not do validation on that topic.